drivers/network/starfive/eth_driver.mk

```diff
@@ -0,0 +1,29 @@
+#
+# Copyright 2024, UNSW
+#
+# SPDX-License-Identifier: BSD-2-Clause
+#
+# Include this snippet in your project Makefile to build
+# the Synopsys dwmac 5.10a NIC driver
+#
+# NOTES:
+#   Generates eth_driver.elf
+#   Needs eth_regs to be set in System Description File
+#   Assumes libsddf_util_debug.a is in LIBS
+
+ETHERNET_DRIVER_DIR := $(dir $(lastword $(MAKEFILE_LIST)))
+
+CHECK_NETDRV_FLAGS_MD5:=.netdrv_cflags-$(shell echo -- ${CFLAGS} ${CFLAGS_network} | shasum | sed 's/ *-//')
+
+${CHECK_NETDRV_FLAGS_MD5}:
+	-rm -f .netdrv_cflags-*
+	touch $@
+
+eth_driver.elf: starfive/ethernet.o
```

Comment:
Probably want a better name for this, as it'll be the same driver for a whole family of dwmac eqos NICs